.github/workflows/scripts/lint-includes.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# Copyright 2020 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import sys
+import pathlib
+import subprocess
+import re
+
+
+def get_modules(root):
+    """
+    Get new-style modules (ie those that use the new layout where public headers
+    are separated from private headers).
+    """
+    modules = (x for x in root.iterdir() if x.is_dir())
```

Comment:
I think you'll need an os.walk or something, for example this won't catch `src/wasm/parser/include`